Adds regression test for subsetting projected grids with a bounding box

flamingbear opened the pull request:
## Description

Adds a test to make a subset request for an ABoVE test granule.
Updates requirements file and lock file

## Jira Issue ID
https://bugs.earthdata.nasa.gov/browse/DAS-1590

## Local Test Steps

 - Run the relevant tests from the notebook SDS_Regression.ipynb locally.
 - Run harmony locally.
 - edit SDS_Regression.ipynb to edit the harmony_host_url  to point to your local harmony
        e.g.  harmony_host_url = 'http://localhost:3000'
 - run the first cells up until the # Begin regression tests: cell
 - skip down to the  "Spatial Subsetting of projected grid." cell
 - Run cells down to MaskFill.
 - note the shapes of the lat/lon grids with respect to the NEE data grid.
 - Reference data is compared in the cell "# compare output data with reference data"






## PR Acceptance Checklist
* [ X ] Acceptance criteria met
* [ X ] Tests added/updated (if needed) and passing
* [ N/A ] Documentation updated (if needed)

owenlittlejohns commented on test/sds/SDS_Regression.ipynb:
```diff
@@ -1046,7 +1045,160 @@
+    "ax = plt.subplot(1, 2, 2, projection=albers_projection)\n",
+    "ax.add_feature(ocean)\n",
+    "ax.add_feature(land)\n",
+    "ax.gridlines(color='grey', linestyle='dashed', draw_labels=True)\n",
```
While I'm maybe leaning towards removing plots in the future - nice call to change the grid line colour. The gray is definitely easier to see against the plotted data (I wish I'd used it in my demo).

owenlittlejohns commented on test/sds/SDS_Regression.ipynb:
```diff
@@ -1046,7 +1045,160 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import xarray as xr\n",
```
Nit: Maybe this import could go at the top of the notebook with the others.

owenlittlejohns commented on test/sds/SDS_Regression.ipynb:
```diff
@@ -1046,7 +1045,160 @@
+    "                                     variables=['NEE', 'lat', 'lon'],\n",
+    "                                     temporal=hoss_projected_info['temporal_range'])\n",
+    "    submit_and_download(harmony_client, hoss_projected_request, hoss_projected_filename)\n",
+    "    assert exists(hoss_projected_filename), 'Unsuccessful HOSS spatial subset request.'\n",
```
The other tests finish up by printing out a green bit of text that says the test was successful. It's not a big deal at all, but I thought it was a nice visual way to see that the test succeeded maybe. (Although in this case that text would likely come after the comparison to the reference image, which is the most important part of this test)

flamingbear replied:
I moved the comparison under the images and added a print_success statement as the last part of this test.

owenlittlejohns reviewed (changes requested):
The test looks good (and it ran locally for me). One thing that isn't tested is a shape file request, which should also be successful 🤞. I think that would be a good addition here. To minimise reference files, that shape file could just contain GeoJSON for a polygon that describes the bounding box already being used.

I'm a _big_ fan of the comparison to the reference images. When we do DAS-1500, I think that is something we should do for all of the tests, potentially also removing a lot of the plotting.

Otherwise, I only really have OCD nitpick comments.

flamingbear commented:
@owenlittlejohns I think I addressed all of the feedback, adding another test for shapefile polygon subsetting that uses the same region as the bounding box did.  It's possible I've made some poor naming decisions at this point so be as nit-picky as you like.

owenlittlejohns reviewed (approved):
Looks good to me. Thanks for adding that extra test and for fixing those minor indentation things (my OCD thanks you for the latter!).

Also - the tests both passed with the latest updates.
